domain/src/Services/TransactionDispatcher/Handlers/IncomeHandler.php

```diff
@@ -28,6 +28,8 @@ public function handle(Transaction $transaction): void
 
         // @phpstan-ignore-next-line
         $taxYearAggregate->recordIncome(new RecordIncome(taxYear: $taxYear, amount: $transaction->marketValue));
+
+        $this->taxYearRepository->save($taxYearAggregate);
```

Comment:
ℹ️ As mentioned earlier, I previously forgot to persist the aggregate in a few places